geo/src/algorithm/rotate.rs

```diff
@@ -121,9 +128,11 @@ where
     T: Float,
 {
     fn rotate(&self, angle: T) -> Self {
-        let pts = vec![self.start, self.end];
-        let rotated = rotation_matrix(angle, self.centroid(), &pts);
-        Line::new(rotated[0], rotated[1])
+        let centroid = self.centroid();
+        Line::new(
+            rotate_one(angle, centroid, self.start),
+            rotate_one(angle, centroid, self.end),
+        )
```

Comment:
Previous implementation resulted in heap allocations with the `vec![...]` creation and also the `Vec` returned from `rotation_matrix`. New implementation shouldn't have any heap allocations